### DISCUSSION TITLE
Pseudotime with small number of cells

### DISCUSSION THREAD

**Discussion by galib36** [GreenleafLab/ArchR#851] [Questions / Documentation]
Hello,
I am trying to do a pseudotime with 4 samples (8_week, 12_week, 14_week_17 week) of a specific cell type (parenchymal cells). One of the sample (12_week) has very few parenchymal cells (27 cells). Taking 5% of this number would be really small, 1.35 cells to be precise. How the pseudo time for this sample is calculated? Did it take only these ~2 cells for pseudo time calculation?

Cells from this sample has a weird distribution as majority of them looks more matured than 14 weeks. I do not know whether this is due to the small number of cells or a real biological phenomenon? Any way to debug this?

**Comment by rcorces** [comment]
The first question is whether the trajectory is even a biologically meaningful idea. I've heard of people forcing trajectories in low dimensional space through samples rather than through clusters. I'm not sure I agree with this approach. The idea behind pseudotime is that it is not actually time and is based on the structure of the data itself.

That being said, I think your interpretation is correct and using so few cells will cause artifacts that are difficult to predict and I dont think there is an easy way to prove it without more data.

**Reply by galib36** [reply]
Thank you @rcorces for your prompt reply. In this case we are interested in parenchymal cell lineages and so plotted it across the samples of different timepoints. We wanted to see whether we can cluster the cells into discrete stages other than the developmental time-point.

Is there any way to control the number of cells during the initial supervised trajectory? Can I change it from 5% to using all the cells?

**Reply by rcorces** [reply]
Sorry if I misunderstood your original question. I dont think your original interpretation is fully correct. From the manual:
"Next, for each cluster, ArchR calculates the mean coordinates for each cell group/cluster in N-dimensions and retains cells whose Euclidean distance to those mean coordinates is in the top 5% of all cells."
So the `preFilterQuantile` and `postFilterQuantile` parameters are acting on all cells, not just the one in your grouping.

**Reply by galib36** [reply]
Thank you so much for the clarification. I missed the all cells bit at the last and interpreted as top 5% of all cells in the group. One more questions, in the `preFilterQuantile` the default value is set to 0.9. But in the manual it says top 5% of the cells, but will it not be top 10% of the cells? or am I missing something here as well?

**Reply by rcorces** [reply]
The documentation should say top 10%. That must be a carryover from what was done in the manuscript (top 5%).

**Reply by galib36** [reply]
Thank you again, that explains all....
